cnxarchive/database.py

```diff
@@ -112,6 +113,34 @@ def initdb(settings):
             for schema_part in get_schema():
                 cursor.execute(schema_part)
 
+            #If virtualenv is active, use that for postgres
+            if hasattr(sys, 'real_prefix'):
+                virtualenv_path = os.path.realpath(sys.prefix)
+                cursor.execute("SELECT current_database();")
+                db_name = cursor.fetchone()[0]
+
+                cursor.execute("""SELECT schema_name FROM
+                                  information_schema.schemata
+                                  WHERE schema_name = 'venv';""")
+                schema_exists = cursor.fetchall()
+
+                if not schema_exists:
+                    cursor.execute("CREATE SCHEMA venv")
+                    cursor.execute("ALTER DATABASE \"{}\" SET "
+                                   "session_preload_libraries ="
+                                   "'session_exec'".format(db_name))
+                    cursor.execute("ALTER DATABASE \"{}\" SET "
+                                   "session_exec.login_name = "
+                                   "'venv.activate_venv'".format(db_name))
+                    cursor.execute("CREATE FUNCTION venv.activate_venv() "
+                                   "RETURNS void LANGUAGE plpythonu AS $_$\n"
+                                   "import os\n"
+                                   "activate_this = os.path.join('{"
+                                   "}/bin/activate_this.py')\n"
+                                   "exec(open(activate_this).read(),"
+                                   "dict(__file__=activate_this))"
+                                   "$_$".format(virtualenv_path))
```

Comment:
This is how I had it, @reedstrm thought this way was more readable. :smiley: 
I don't care either way, but to conform to PEP8, it will still need to wrap lines using heredoc style.